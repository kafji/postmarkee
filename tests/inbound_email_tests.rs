use postmarkee::InboundEmail;

#[test]
fn test_deserialize() {
    let serialized = include_str!("./postmark_inbound_payload_example.json");

    let email: InboundEmail = serde_json::from_str(serialized).unwrap();

    insta::assert_debug_snapshot!(email);
}

#[test]
fn test_serialize() {
    let email: InboundEmail = {
        let serialized = include_str!("./postmark_inbound_payload_example.json");
        serde_json::from_str(serialized).unwrap()
    };

    let serialized = serde_json::to_string_pretty(&email).unwrap();

    insta::assert_display_snapshot!(serialized);
}
