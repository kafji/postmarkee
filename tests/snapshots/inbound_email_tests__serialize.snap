---
source: tests/inbound_email_tests.rs
assertion_line: 25
expression: serialized

---
{
  "FromName": "Postmarkapp Support",
  "MessageStream": "inbound",
  "FromFull": {
    "Email": "support@postmarkapp.com",
    "Name": "Postmarkapp Support",
    "MailboxHash": ""
  },
  "ToFull": [
    {
      "Email": "yourhash+SampleHash@inbound.postmarkapp.com",
      "Name": "Firstname Lastname",
      "MailboxHash": "SampleHash"
    }
  ],
  "CcFull": [
    {
      "Email": "firstcc@postmarkapp.com",
      "Name": "First Cc",
      "MailboxHash": ""
    },
    {
      "Email": "secondCc@postmarkapp.com",
      "Name": "",
      "MailboxHash": ""
    }
  ],
  "BccFull": [
    {
      "Email": "firstbcc@postmarkapp.com",
      "Name": "First Bcc",
      "MailboxHash": ""
    },
    {
      "Email": "secondbcc@postmarkapp.com",
      "Name": "",
      "MailboxHash": ""
    }
  ],
  "OriginalRecipient": "yourhash+SampleHash@inbound.postmarkapp.com",
  "Subject": "Test subject",
  "MessageID": "73e6d360-66eb-11e1-8e72-a8904824019b",
  "ReplyTo": "replyto@postmarkapp.com",
  "MailboxHash": "SampleHash",
  "Date": "Fri, 1 Aug 2014 20:45:32 +00:00",
  "TextBody": "This is a test text body.",
  "HtmlBody": "<html><body><p>This is a test html body.</p></body></html>",
  "StrippedTextReply": "This is the reply text",
  "Tag": "TestTag",
  "Headers": [
    {
      "Name": "X-Header-Test",
      "Value": ""
    },
    {
      "Name": "X-Spam-Status",
      "Value": "No"
    },
    {
      "Name": "X-Spam-Score",
      "Value": "-0.1"
    },
    {
      "Name": "X-Spam-Tests",
      "Value": "DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,SPF_PASS"
    }
  ],
  "Attachments": [
    {
      "Name": "test.txt",
      "Content": "VGhpcyBpcyBhdHRhY2htZW50IGNvbnRlbnRzLCBiYXNlLTY0IGVuY29kZWQu",
      "ContentType": "text/plain",
      "ContentLength": 45
    }
  ]
}
